salt/modules/file.py

```diff
@@ -1910,103 +1912,79 @@ def line(path, content=None, match=None, mode=None, location=None,
                 _assert_occurrence(body, after, 'after')
                 out = []
                 lines = body.split(os.linesep)
-                for idx in range(len(lines)):
-                    _line = lines[idx]
+                for idx, _line in enumerate(lines):
                     out.append(_line)
                     cnd = _get_line_indent(_line, content, indent)
-                    if _line.find(after) > -1:
-                        # No dupes or append, if "after" is the last line
-                        if (idx < len(lines) and _starts_till(lines[idx + 1], cnd) < 0) or idx + 1 == len(lines):
-                            out.append(cnd)
+                    # No duplicates or append, if "after" is the last line
+                    if (_line.find(after) > -1 and
+                            (lines[((idx + 1) < len(lines)) and idx + 1 or idx].strip() != cnd or
+                             idx + 1 == len(lines))):
+                        out.append(cnd)
                 body = os.linesep.join(out)
 
         else:
             if location == 'start':
-                body = ''.join([content, body])
+                body = os.linesep.join([content, body])
```

Comment:
Using a tuple instead of a list in the join is quicker, just FYI.

```python
>>> import timeit
>>> timeit.Timer('"".join(["foo", "bar"])').repeat()
[0.15700793266296387, 0.12559294700622559, 0.12819194793701172]
>>> timeit.Timer('"".join(("foo", "bar"))').repeat()
[0.11858606338500977, 0.08025217056274414, 0.07921886444091797]
```